kotest-property/src/jvmMain/kotlin/io/kotest/property/arbitrary/bigdecimal.kt

```diff
@@ -4,9 +4,22 @@ import io.kotest.property.Arb
 import java.math.BigDecimal
 import java.math.RoundingMode
 
-fun Arb.Companion.bigDecimal() = Arb.double().map { it.toBigDecimal() }
+private val bigDecimalEdgecases = listOf(
+   BigDecimal(0.0),
+   BigDecimal(1.0),
+   BigDecimal(-1.0),
+   BigDecimal("1e-300"),
+   BigDecimal("-1e-300"),
+)
 
-fun Arb.Companion.bigDecimal(scale: Int, roundingMode: RoundingMode) = bigDecimal().map { it.setScale(scale, roundingMode) }
+fun Arb.Companion.bigDecimal(): Arb<BigDecimal> {
+   return arbitrary(bigDecimalEdgecases) {
+      BigDecimal(it.random.nextInt() * it.random.nextDouble())
```

Comment:
Perhaps we should do BigDecimal(it.random.nextDouble()).multiply(BigDecimal(it.random.nextLong()) to allow for actual big decimals.

Reply:
Doing this way is always produce quite a larger digit BigDecimal `5494701162849313711.73708734` to get smaller BigDecimal can combine both nextInt and nextLong.

```kotlin
fun Arb.Companion.bigDecimal(): Arb<BigDecimal> {
   return arbitrary(bigDecimalEdgecases) {
      val intOrLong = if (it.random.nextInt() % 2 == 0) it.random.nextLong() else it.random.nextInt().toLong()
      BigDecimal(it.random.nextDouble()).multiply(BigDecimal(intOrLong))
   }
}
```